go/consensus/cometbft/apps/keymanager/churp/txs.go

```diff
@@ -44,6 +47,11 @@ func (ext *churpExt) create(ctx *tmapi.Context, req *churp.CreateRequest) error
 		return fmt.Errorf("keymanager: churp: invalid signer")
 	}
 
+	// Add stake claim to the key manager owner.
+	if err = addStakeClaim(ctx, kmRt.EntityID, req.RuntimeID, req.ID); err != nil {
```

Comment:
This should be done after all validation is done (you have a TODO below on line 86). Also since you are now updating multiple things you should start a storage transaction to ensure atomicity, e.g.:
```golang
// Start a new transaction and rollback in case we fail.
ctx = ctx.NewTransaction()
defer ctx.Close()
// ...
ctx.Commit()
```

Reply:
Yeah. So in practice in those cases a failure would mean that something catastrophic has happened and the node would not be able to continue anyway. But it makes sense to be safe.